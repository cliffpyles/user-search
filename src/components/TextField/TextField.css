.TextField input {
  padding: 0.5rem 1rem;
  border: var(--TextField--Border, 1px solid #e1e1e1);
  background: var(--TextField--Background, #fff);
  color: var(--TextField--Color, #333);
  font-family: var(--TextField--FontFamily, system-ui);
  font-size: var(--TextField--FontSize, 16px);
}
