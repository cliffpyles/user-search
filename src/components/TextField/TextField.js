import React from 'react'
import './TextField.css'

export default function TextField({
  id,
  label,
  name,
  onChange,
  type = 'text',
  value
}) {
  return (
    <div className="TextField">
      <label htmlFor={id}>{label}</label>
      <input
        id={id}
        name={name}
        onChange={onChange}
        type={type}
        value={value}
      />
    </div>
  )
}
