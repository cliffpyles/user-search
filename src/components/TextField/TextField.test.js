import React from 'react'
import Renderer from 'react-test-renderer'
import TextField from './'

test('can render a TextField', () => {
  const component = Renderer.create(
    <TextField id="example" name="example" label="example" value="woohoo!" />
  )

  expect(component.toJSON()).toMatchSnapshot()
})
