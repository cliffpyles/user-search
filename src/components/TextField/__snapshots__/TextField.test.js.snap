// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render a TextField 1`] = `
<div
  className="TextField"
>
  <label
    htmlFor="example"
  >
    example
  </label>
  <input
    id="example"
    name="example"
    type="text"
    value="woohoo!"
  />
</div>
`;
